### PR TITLE
New controls

### PR DESCRIPTION
**Pull request by hbons**
Implement the new component library with new converters.

### CODE REVIEW

**Comment by hbons** [comment]
don't worry, I'll redo all the commits so they make sense. just backing up. :)

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/CocoaConverter.cs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+		}
+
+
+		protected nfloat GetNSFontWeight(FigmaText text)
+		{
+			string weight = text?.style?.fontPostScriptName;
+
+			if (weight != null)
+			{
+				if (weight.EndsWith("-Black"))
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/CocoaConverter.cs:170]: all this switch can be replaced by    

var index = name.LastIndexOf ("-");
            if (index > -1) {
                var memberInfo = typeof(NSFontWeight).GetMember(name.Substring(index + 1));
                if (memberInfo.GetValue (0) is PropertyInfo propertyInfo)
                    return (nfloat)propertyInfo.GetValue(null);
            }

**Comment by netonjm** [comment]
Please we need to avoid this kind of PR's, this is impossible to review and can provoke a lot of unexpected regressions.

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/ComboBoxConverter.cs [hunk 2]
```diff
@@ -38,90 +38,88 @@
...
 		}
 
+
 		protected override StringBuilder OnConvertToCode(FigmaCodeNode currentNode, FigmaCodeNode parentNode, FigmaCodeRendererService rendererService)
 		{
-			var figmaInstance = (FigmaFrameEntity)currentNode.Node;
-
-			var builder = new StringBuilder ();
-			var name = Resources.Ids.Conversion.NameIdentifier;
+			var code = new StringBuilder ();
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/ComboBoxConverter.cs:83]: codeStringBuilder or codeBuilder or builder is more descriptive about what kind of object we are working on

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/ColorWellConverter.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			var colorWell = new NSColorWell();
+			colorWell.Color = NSColor.ControlAccentColor;
+
+			return new View(colorWell);
+		}
+
+
+		protected override StringBuilder OnConvertToCode(FigmaCodeNode currentNode, FigmaCodeNode parentNode, FigmaCodeRendererService rendererService)
+		{
+			var code = new StringBuilder();
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/ColorWellConverter.cs:61]: codeStringBuilder or codeBuilder or builder is more descriptive about what kind of object we are working on

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/DisclosureConverter.cs [hunk 4]
```diff
@@ -40,62 +37,79 @@
...
-			instance.TryGetNativeControlComponentType (out var controlType);
-			switch (controlType) {
-				case NativeControlComponentType.DisclosureTriangleStandard:
-				case NativeControlComponentType.DisclosureTriangleStandardDark:
-					nativeView.ControlSize = NSControlSize.Regular;
-					break;
-			}
+			var disclosureView = new NSView();
+
+			var subView = new NSView(disclosureView.Bounds);
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/DisclosureConverter.cs:57]: Frame is not taking into account in Constraint world

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/DisclosureConverter.cs [hunk 5]
```diff
@@ -40,62 +37,79 @@
...
+			subView.WantsLayer = true;
+		    subView.Layer.BackgroundColor = NSColor.Red.CGColor;
+
+
+			var disclosureTriangle = new NSButton();
+			disclosureTriangle.SetFrameSize(new CoreGraphics.CGSize(130, 13));
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/DisclosureConverter.cs:63]: this needs to be changed as a constraint

File: FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/DisclosureConverter.cs [hunk 6]
```diff
@@ -40,62 +37,79 @@
...
+			disclosureTriangle.SetButtonType(NSButtonType.PushOnPushOff);
+			disclosureTriangle.BezelStyle = NSBezelStyle.Disclosure;
 
-			return view;
+            disclosureTriangle.Activated += delegate {
```
Comment (netonjm) [inline FigmaSharp.NativeControls/FigmaSharp.NativeControls.Cocoa/Converters/DisclosureConverter.cs:67]: empty delegate

**Pull request closed without merging**